tests/unit/members-dao.test.js

```diff
@@ -0,0 +1,349 @@
+  });
+
+  describe('Test postMember', () => {
+    it('Improper body should be rejected', () => {
+      createConnStub();
+
+      const result = membersDao.postMember({});
+      return result.should
+        .eventually.be.rejected
+        .and.be.an.instanceOf(TypeError);
+    });
+
+    it('Should be fulfilled with single result', () => {
+      const expectedResult = {};
+      createConnStub(testCases.singleResultWithOutId);
+
+      const result = membersDao.postMember(fakeMemberPostBody);
+      return result.should
+        .eventually.be.fulfilled
+        .and.deep.equal(expectedResult);
+    });
+
+    it('Should be rejected when there are no OutId in the result field', () => {
+      createConnStub(testCases.singleResult);
+
+      const result = membersDao.postMember(fakeMemberPostBody);
+      return result.should
+        .eventually.be.rejected
+        .and.be.an.instanceOf(TypeError);
+    });
+  });
+
+  describe('Test deleteMember', () => {
+    const validTestCases = [
+      testCases.singleResult,
+      testCases.emptyResult,
+    ];
+    _.forEach(validTestCases, (testCase) => {
+      it(`deleteMember should be fulfilled with ${testCase.description}`, () => {
+        createConnStub(testCase);
+
+        const expectedResult = testCase.data;
+        const result = membersDao.deleteMember(fakeId);
+        return result.should
+          .eventually.be.fulfilled
+          .and.deep.equal(expectedResult)
+          .then(() => {
+
+          });
+      });
+    });
+  });
+
+  describe('Test patchMember', () => {
+    const validFalseTestCases = [
+      testCases.singleResult,
+      testCases.emptyResult,
+      testCases.rowEffectZeroResult,
+    ];
+
+    it('Empty body should be fulfilled while the execute function from conn object is not called at all', () => {
+      const connStub = createConnStub();
+
+      const result = membersDao.patchMember(fakeId, {});
+      return result.should
+        .eventually.be.fulfilled
+        .and.equal(true)
+        .then(() => {
+          sinon.assert.callCount(connStub.executeStub, 0);
+        });
+    });
+
+    _.forEach(validFalseTestCases, (testCase) => {
+      it(`patchMember should be fulfilled with ${testCase.description} and return false`, () => {
+        createConnStub(testCase);
+
+        const expectedResult = false;
+        const result = membersDao.patchMember(fakeId, fakeMemberPatchBody);
+        return result.should
+          .eventually.be.fulfilled
+          .and.deep.equal(expectedResult)
+          .then(() => {
+
+          });
+      });
+    });
+
+    it('patchMember should be fulfilled with one row effected result and return true', () => {
+      createConnStub(testCases.rowEffectedOneResult);
+
+      const expectedResult = true;
+      const result = membersDao.patchMember(fakeId, fakeMemberPatchBody);
+      return result.should
+        .eventually.be.fulfilled
+        .and.deep.equal(expectedResult)
+        .then(() => {
+
```

Comment:
There's a few cases like this where it looks like there's some missing logic. Could you either add something here or remove the `.then()`?